src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -24,11 +24,103 @@
+import org.spongepowered.api.Sponge;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
 
+    /**
+     * Returns the {@link RailDirection} from the given facing directions.
+     *
+     * @param firstDirection The first direction
+     * @param secondDirection The second direction
+     * @return The {@link RailDirection}
+     * @throws IllegalArgumentException If the given directions don't differ
+     */
+    static RailDirection fromFaces(final Direction firstDirection, final Direction secondDirection) {
+        checkNotNull(firstDirection, "firstDirection");
+        checkNotNull(secondDirection, "secondDirection");
+
+        Direction cardinalFirstDirection = Direction.getClosest(firstDirection.asOffset(), Direction.Division.CARDINAL);
+        Direction cardinalSecondDirection = Direction.getClosest(secondDirection.asOffset(), Direction.Division.CARDINAL);
+
+        if (cardinalFirstDirection.equals(cardinalSecondDirection)) {
+            throw new IllegalArgumentException("The directions should be different!");
+        }
+
+        for (RailDirection currentRailDirection: Sponge.getRegistry().getAllForMinecraft(RailDirection.class)) {
+            if (currentRailDirection.isFacing(cardinalFirstDirection) && currentRailDirection.isFacing(cardinalSecondDirection)) {
+                return currentRailDirection;
+            }
+        }
+
+        throw new AssertionError();
+    }
+
+    /**
+     * Returns the {@link RailDirection} from the given ascending direction.
+     *
+     * @param ascendingDirection The ascending direction
+     * @return The {@link RailDirection}
+     */
+    static RailDirection fromAscendingDirection(final Direction ascendingDirection) {
+        checkNotNull(ascendingDirection, "ascendingDirection");
+
+        Direction cardinalAscendingDirection = Direction.getClosest(ascendingDirection.asOffset(), Direction.Division.CARDINAL);
+
+        for (RailDirection currentRailDirection: Sponge.getRegistry().getAllForMinecraft(RailDirection.class)) {
+            Optional<Direction> currentAscendingDirectionBox = currentRailDirection.getAscendingDirection();
+
+            if (currentAscendingDirectionBox.isPresent()) {
+                Direction currentAscendingDirection = currentAscendingDirectionBox.get();
+
+                if (currentAscendingDirection.equals(cardinalAscendingDirection)) {
+                    return currentRailDirection;
+                }
+            }
+        }
+
+        throw new AssertionError();
+    }
+
+    /**
+     * Returns the direction it's ascending to, if this rail direction is ascending.
+     *
+     * @return The ascending direction, if possible
+     */
+    Optional<Direction> getAscendingDirection();
```

Comment:
Should we just update `Direction` to make it a true 3D directional enum (so, `UP_NORTH` for example) and do away with this method? Alternatively, should we have a separate `RailAxis` instead that have a combination that contains a combination of `Direction`s?